crates/router/src/services/api.rs

```diff
@@ -372,7 +372,13 @@ where
                     let response = call_connector_api(state, request).await;
                     let external_latency = current_time.elapsed().as_millis();
                     logger::debug!(connector_response=?response);
-
+                    let status_code = response
+                        .as_ref()
+                        .map(|i| {
+                            i.clone()
```

Comment:
Is clone needed here? Since status_code is a u8 and can be copied directly...

Reply:
it becomes *i from ownership of response been transferred